app/models/grda_warehouse/client_file.rb

```diff
@@ -233,15 +234,21 @@ class ClientFile < GrdaWarehouse::File
     ####################
     # Callbacks
     ####################
-    after_create_commit :notify_users, if: ->(m) { m.should_run_callbacks? }
-    before_save :adjust_consent_date, if: ->(m) { m.should_run_callbacks? }
-    after_save :note_changes_in_consent, if: ->(m) { m.should_run_callbacks? }
-    after_commit :set_client_consent, on: [:create, :update], if: ->(m) { m.should_run_callbacks? }
+    after_create_commit :notify_users, if: ->(m) { m.should_run_callbacks? } # rubocop:disable Style/SymbolProc
+    before_update :adjust_revoked_by, if: ->(m) { m.should_run_callbacks? && consent_revoked_at_changed? }
```

Comment:
Is there some reason we can't just call this from the controller?  There are a bunch of callbacks, but the controller should know the user that made the change, so should be able to set `consent_revoked_by_user_id: current_user.id` when creating or updating the record.  (There is significant complexity here, so I may be missing something.)

While we're in here, it might be worth adding `has_paper_trail`.  I was going to suggest we pull the user out of the versions table, but it appears these aren't being logged.

Reply:
It was mentioned in the ticket that this may be a good way to go. At the time, it felt like capturing it at a deeper level may good if we are working with files in multiple controllers, but I see what you are saying and it does feel less complex to just have the controller handle it. I'll adjust the PR.